docs/guides/ssr.md

```diff
@@ -117,6 +117,175 @@ There's a catch if you're using [Next.js' rewrites feature](https://nextjs.org/d
 
 The result is missing referential equality for all the hydration data, which for example triggers wherever your data is used as props of components or in the dependency array of `useEffect`s/`useMemo`s.
 
+## Using Experimental `app` Directory in Next.js 13
+
+> **WARNING:** The `app` directory introduced in Next.js 13 is currently in beta, and it is not recommended for use in production. The API is not stable, and the guidelines provided here may become obsolete in a future update. This guide is provided as is to supply a quick start for early exploration of Next.js 13's experimental features.
+
+Both prefetching approaches, using `initialData` or `<Hydrate>`, are available within the `app` directory.
+
+- Prefetch the data in a Server Component and prop drill `initialData` to Client Components
+  - Quick to set up for simple cases
+  - May need to prop drill through multiple layers of Client Components
+  - May need to prop drill to multiple Client Components using the same query
+  - Query refetching is based on when the page loads instead of when the data was prefetched on the server
+- Prefetch the query on the server, dehydrate the cache and rehydrate it on the client with `<Hydrate>`
+  - Requires slightly more setup up front
+  - No need to prop drill
+  - Query refetching is based on when the query was prefetched on the server
+
+
+### `<QueryClientProvider>` is required by both the `initialData` and `<Hydrate>` prefetching approaches
+
+The hooks provided by the `react-query` package need to retrieve a `QueryClient` from their context. Wrap your component tree with `<QueryClientProvider>` and pass it an instance of `QueryClient`.
+
+```tsx
+// app/providers.jsx
+'use client'
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+
+export default function Providers({children}) {
+  const [queryClient] = React.useState(() => new QueryClient())
+  
+  return (
+    <QueryClientProvider client={queryClient}>
+      {children}
+    </QueryClientProvider>
+  )
+}
+```
+
+```tsx
+// app/layout.jsx
+import Providers from './providers'
+
+export default function RootLayout({children}) {
+  return (
+    <html>
+      <head></head>
```

Comment:
Good catch 🕺

Reply:
wow, big TIL, @efilion. Thanks for that!

It still looks to me that even when that change lands, it will be a workaround based on specific use-cases. If it's not necessary, I'd still suggest to omit it.

You make a good point about `create-next-app` though. I'm not aware of **AppDir** as part of the CLI template though. I'd agree 100% in keeping it consistent with the default template if there's already one.